.buildkite/premerge.steps.yaml

```diff
@@ -0,0 +1,26 @@
+---
+agent_transients: &agent_transients
+  # This is designed to trap and retry failures because agent lost
+  # connection. Agent exits with -1 in this case.
+  exit_status: -1
+  limit: 3
+  
+common: &common
+  agents:
+    - "capable_of_building=gdk-for-unreal"
+    - "environment=production"
+    - "permission_set=pusher"
+    - "queue=v2-1549893371-183fd9b7177e2649-------z"
```

Comment:
Does this queue have any versioning info against it? Is it easy to validate what changes this image has?

Reply:
Yeah, the format is `v2-<timestamp>-<git_hash>-------z` (where `-------z` is a set of state markers that denote the local state of the repo at the time the build was run.

You can always look at <git_hash> of the platform repo to confirm the exact configuration that was used to build that agent, and the timestamp can be useful for determining the historical state of any external dependencies that can't be pinned (though obviously, using a specific pinned version of your dependencies is vastly preferable for proving/reconstructing state).

Reply:
Is it worth adding a comment that briefly summarises that?

Reply:
We probably ought to have a page outlining this on http://brevi.link/buildkite-docs.  It hasn't been needed in practice outside of the EV team so far, but it's valuable to know that you _can_ do it.  

I've filed https://improbableio.atlassian.net/browse/ENG-1319 for us to add that to the documentation.